great_expectations/execution_engine/sqlalchemy_execution_engine.py

```diff
@@ -1032,6 +1004,175 @@ def close(self) -> None:
+        query entry, then resets the accumulation state for the next parameter batch.
+
+        This function mutates the domain_batches,
+        batch_counters, and domain_kwargs_map dictionaries.
+        """
+        if domain_id in domain_batches and domain_batches[domain_id]["select"]:
+            batch_idx = batch_counters.get(domain_id, 0)
+            domain_kwargs = domain_kwargs_map[domain_id]
+
+            if batch_idx == 0:
+                final_domain_id = domain_id
+            else:
+                final_domain_id = IDDict({**domain_kwargs, "_batch_idx": batch_idx}).to_id()
+
+            queries[final_domain_id] = {
+                "select": domain_batches[domain_id]["select"],
+                "metric_ids": domain_batches[domain_id]["metric_ids"],
+                "domain_kwargs": domain_kwargs,
+            }
+
+            domain_batches[domain_id] = {"select": [], "metric_ids": []}
+            batch_counters[domain_id] = batch_idx + 1
+
+    def _organize_metrics_with_parameter_limit(
+        self, metric_fn_bundle: Iterable[MetricComputationConfiguration]
+    ) -> dict[IDDictID, dict]:
+        """Organize metrics from bundle into queries while respecting parameter limits.
+
+        Args:
+            metric_fn_bundle: Iterable of metric configurations to organize into queries.
+
+        Returns:
+            Dictionary of domain IDs mapped to query configurations
+            with select expressions and metric IDs.
+        """
+        MAX_PARAMS_PER_QUERY = 256
+        queries: dict[IDDictID, dict] = {}
+        domain_batches: dict[IDDictID, dict] = {}
+        batch_counters: dict[IDDictID, int] = {}
+        domain_kwargs_map: dict[IDDictID, dict] = {}
+
+        for bundled_metric_configuration in metric_fn_bundle:
+            metric_to_resolve: MetricConfiguration = (
+                bundled_metric_configuration.metric_configuration
+            )
+            metric_fn: Any = bundled_metric_configuration.metric_fn
+            domain_kwargs: dict = bundled_metric_configuration.compute_domain_kwargs or {}
+            if not isinstance(domain_kwargs, IDDict):
+                domain_kwargs = IDDict(domain_kwargs)
+
+            domain_id = domain_kwargs.to_id()
+            selectable: sqlalchemy.Selectable = self.get_domain_records(domain_kwargs=domain_kwargs)
+
+            if domain_id not in domain_batches:
+                domain_batches[domain_id] = {"select": [], "metric_ids": []}
+                batch_counters[domain_id] = 0
+                domain_kwargs_map[domain_id] = domain_kwargs
+
+            test_selects = domain_batches[domain_id]["select"] + [
+                metric_fn.label(metric_to_resolve.metric_name)
+            ]
+            test_param_count = self._count_query_parameters(selectable, test_selects)
+
+            if test_param_count > MAX_PARAMS_PER_QUERY and domain_batches[domain_id]["select"]:
```

Comment:
Should this be happening only for databricks?

Reply:
Does this work if we have, say 600 parameters. Unless I'm not following the code correctly (to be honest, it's pretty hard to follow), it looks like we would only ever add one additional query

Reply:
I added a test to hit around 600 metrics (parameters). Let me know if you want to chat to discuss the whole logic, if it still doesn't make sense.

Reply:
Good idea!